notebooks/advanced/Time_Series_Forecasting_with_Unstructured_Data_and_Amazon_SageMaker_Neural_Topic_Model/3_Forecast.ipynb

```diff
@@ -0,0 +1,972 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define the schema for the related time series"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set the dataset name to a new unique value. If it already exists, go to the Forecast console and delete any existing\n",
+    "# dataset ARNs and datasets.\n",
+    "\n",
+    "datasetName = 'webtraffic_forecast_related_NLP'\n",
+    "schema ={\n",
+    "   \"Attributes\":[{\n",
+    "         \"AttributeName\":\"item_id\",\n",
+    "         \"AttributeType\":\"string\"\n",
+    "      }, \n",
+    "       {\n",
+    "         \"AttributeName\":\"timestamp\",\n",
+    "         \"AttributeType\":\"timestamp\"\n",
+    "      },\n",
+    "       {\n",
+    "         \"AttributeName\":\"SentimentHeadline\",\n",
+    "         \"AttributeType\":\"float\"\n",
+    "      }]\n",
+    "    + \n",
+    "      [{\n",
+    "         \"AttributeName\":\"Headline_{}\".format(x),\n",
+    "         \"AttributeType\":\"float\"\n",
+    "      } for x in range(10)] \n",
+    "}\n",
+    "\n",
+    "try:\n",
+    "    response=forecast.create_dataset(\n",
+    "                    Domain=\"WEB_TRAFFIC\",\n",
+    "                    DatasetType='RELATED_TIME_SERIES',\n",
+    "                    DatasetName=datasetName,\n",
+    "                    DataFrequency=DATASET_FREQUENCY, \n",
+    "                    Schema = schema\n",
+    "                   )\n",
+    "    related_datasetArn = response['DatasetArn']\n",
+    "    print('Success')\n",
+    "except Exception as e:\n",
+    "    print(e)\n",
+    "    related_datasetArn = 'arn:aws:forecast:{}:{}:dataset/{}'.format(REGION_NAME, ACCOUNT_NUM, datasetName)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define the dataset group"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before ingesting any data into Forecast we need to combine the target and related time series into a dataset group. We define this below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datasetGroupName = 'webtraffic_forecast_NLPgroup'\n",
+    "    \n",
+    "try:\n",
+    "    create_dataset_group_response = forecast.create_dataset_group(DatasetGroupName=datasetGroupName,\n",
+    "                                                              Domain=\"WEB_TRAFFIC\",\n",
+    "                                                              DatasetArns= [datasetArn, related_datasetArn]\n",
+    "                                                             )\n",
+    "    datasetGroupArn = create_dataset_group_response['DatasetGroupArn']\n",
+    "    print('Success')\n",
+    "\n",
+    "except Exception as e:\n",
```

Comment:
Catching generic "Exception" class to skip creating existing resources, but shouldn't do this as it is possible to catch different types of errors.
Use forecast.exceptions.ResourceAlreadyExistsException specifically.